kernel/src/common/deferred_call_mux.rs

```diff
@@ -0,0 +1,207 @@
+//! Hardware-independent kernel interface for deferred calls
+//!
+//! This allows any struct in the kernel which implements
+//! [DeferredCallMuxClient](crate::common::deferred_call_mux::DeferredCallMuxClient)
+//! to set and receive deferred calls.
+//!
+//! These are scheduled "interrupts" in the chip scheduler.
+//! They are especially important if some hardware doesn't
+//! support real interrupts where they are needed, or for
+//! implementing software devices that are supposed to work
+//! like hardware devices.
+//!
+//! Usage
+//! -----
+//!
+//! The `deferred_call_mux_clients` array size determines how many
+//! [DeferredCallHandle](crate::common::deferred_call_mux::DeferredCallHandle)s
+//! may be registered with the Mux.
+//! When no more slots are available,
+//! `deferred_call_mux.register(some_client)` will return `None`.
+//!
+//! ```
+//! # use core::cell::Cell;
+//! # use kernel::common::cells::OptionalCell;
+//! # use kernel::static_init;
+//! use kernel::common::deferred_call_mux::{
+//!     DeferredCallMux,
+//!     DeferredCallMuxClient,
+//!     set_global_mux,
+//! };
+//!
+//! let deferred_call_mux_clients = unsafe { static_init!(
+//!     [(Cell<bool>, OptionalCell<&'static DeferredCallMuxClient>); 1],
+//!     [(Cell::new(false), OptionalCell::empty())]
```

Comment:
Initializing the internal data structure separately seems rather inelegant. Is there some more idiomatic way to do this?

Reply:
You could maybe replace the `tuple (Cell, OptionCell)` with a struct and impl `Default` for it.

However, personally I think `OptionCell` should impl `Default` as `None` then you could `Default::default()` for them

Reply:
> You could maybe replace the `tuple (Cell, OptionCell)` with a struct and impl `Default` for it.

Done.

> However, personally I think `OptionCell` should impl `Default` as `None` then you could `Default::default()` for them

Sounds like a great idea. I've seen you already created a PR for that. :+1: